govcd/user.go

```diff
@@ -0,0 +1,456 @@
+	if err != nil {
+		return nil, fmt.Errorf("failure to retrieve a new user after %s : %s", elapsed, err)
+	}
+
+	return newUser, nil
+}
+
+// SimpleCreateUser creates an org user from a simplified structure
+func (adminOrg *AdminOrg) SimpleCreateUser(userData OrgUserConfiguration) (*OrgUser, error) {
+
+	if userData.Name == "" {
+		return nil, fmt.Errorf("name is mandatory to create a user")
+	}
+	if userData.Password == "" {
+		return nil, fmt.Errorf("password is mandatory to create a user")
+	}
+	if userData.RoleName == "" {
+		return nil, fmt.Errorf("role is mandatory to create a user")
+	}
+	role, err := adminOrg.GetRole(userData.RoleName)
+	if err != nil {
+		return nil, fmt.Errorf("error finding a role named %s", userData.RoleName)
+	}
+
+	var userConfiguration = types.User{
+		Xmlns:           types.XMLNamespaceVCloud,
+		Type:            types.MimeAdminUser,
+		ProviderType:    userData.ProviderType,
+		Name:            userData.Name,
+		IsEnabled:       userData.IsEnabled,
+		Password:        userData.Password,
+		DeployedVmQuota: userData.DeployedVmQuota,
+		StoredVmQuota:   userData.StoredVmQuota,
+		FullName:        userData.FullName,
+		EmailAddress:    userData.EmailAddress,
+		Description:     userData.Description,
+		Role:            &types.Reference{HREF: role.HREF},
+	}
+
+	// ShowUser(userConfiguration)
+	return adminOrg.CreateUser(&userConfiguration)
+}
+
+// GetRoleName retrieves the name of the role currently assigned to the user
+func (user *OrgUser) GetRoleName() string {
+	if user.User.Role == nil {
+		return ""
+	}
+	return user.User.Role.Name
+}
+
+// delete removes the user, returning an error if the call fails.
+// if requested, it will attempt to take ownership before the removal.
+// API Documentation: https://code.vmware.com/apis/442/vcloud-director#/doc/doc/operations/DELETE-User.html
+// Note: in the GUI we need to disable the user before deleting.
+// There is no such constraint with the API.
+func (user *OrgUser) delete(takeOwnership bool) error {
+	util.Logger.Printf("[TRACE] Deleting user: %#v", user.User.Name)
+
+	if takeOwnership {
+		err := user.TakeOwnership()
+		if err != nil {
+			return err
+		}
+	}
+
+	userHREF, err := url.ParseRequestURI(user.User.Href)
+	if err != nil {
+		return fmt.Errorf("error getting HREF for user %s : %v", user.User.Name, err)
+	}
+	util.Logger.Printf("[TRACE] Url for deleting user : %#v and name: %s", userHREF, user.User.Name)
+
+	return user.client.ExecuteRequestWithoutResponse(userHREF.String(), http.MethodDelete,
+		types.MimeAdminUser, "error deleting user : %s", nil)
+}
+
+// UnconditionalDelete deletes the user, WITHOUT running a call to take ownership of the user objects
+// This call will fail if the user has owns *running* vApps/VMs
+func (user *OrgUser) UnconditionalDelete() error {
+	return user.delete(false)
+}
+
+// SafeDelete deletes the user after taking ownership of its objects
```

Comment:
More elaborated comment needed -> that runing code user will take ownership, not deleted one or somehow else

Reply:
added

Reply:
nice